Packages/ChatSecure/ChatSecure/Classes/Networking/Services/User/UserService.swift

```diff
@@ -0,0 +1,290 @@
+	}
+	
+	public func searchUser(keyword: String, domain: String) async -> (Result<User_SearchUserResponse, Error>) {
+		var request = User_SearchUserRequest()
+		request.keyword = keyword
+		
+		return await channelStorage.getChannel(domain: domain).searchUser(request)
+	}
+	
+	public func getUserInfo(clientID: String, workspaceDomain: String, domain: String) async -> (Result<User_UserInfoResponse, Error>) {
+		var request = User_GetUserRequest()
+		request.clientID = clientID
+		request.workspaceDomain = workspaceDomain
+		
+		return await channelStorage.getChannel(domain: domain).getUserInfo(request)
+	}
+	
+	public func getListStatus(data: [[String: String]], domain: String) async -> (Result<User_GetClientsStatusResponse, Error>) {
+		var request = User_GetClientsStatusRequest()
+		let clients = data.map { item -> User_MemberInfoRequest in
+			var client = User_MemberInfoRequest()
+			client.clientID = item["id"] ?? ""
+			client.workspaceDomain = item["domain"] ?? ""
+			return client
+		}
+		request.lstClient = clients
+		request.shouldGetProfile = true
+		
+		return await channelStorage.getChannel(domain: domain).getClientsStatus(request)
+		
+	}
+	
+	public func searchUserWithEmail(emailHash: String, domain: String) async -> (Result<User_FindUserByEmailResponse, Error>) {
+		var request = User_FindUserByEmailRequest()
+		request.emailHash = emailHash
+		
+		return await channelStorage.getChannel(domain: domain).searchUserWithEmail(request)
+	}
+	
+	public func pingRequest(domain: String) async -> Result<User_BaseResponse, Error> {
+		let request = User_PingRequest()
+		return await channelStorage.getChannel(domain: domain).pingRequest(request)
+	}
+	
+	public func updateStatus(status: String, domain: String) async -> Result<User_BaseResponse, Error> {
+		var request = User_SetUserStatusRequest()
+		request.status = status
+		return await channelStorage.getChannel(domain: domain).updateStatus(request)
+	}
+	
+	public func uploadAvatar(fileName: String, fileContentType: String, fileData: Data, fileHash: String, domain: String) async -> (Result<User_UploadAvatarResponse, Error>) {
+		var request = User_UploadAvatarRequest()
+		request.fileName = fileName
+		request.fileContentType = fileContentType
+		request.fileData = fileData
+		request.fileHash = fileHash
+		
+		return await channelStorage.getChannel(domain: domain).uploadAvatar(request)
+	}
+	
+	public func updateProfile(displayName: String, avatar: String, phoneNumber: String, clearPhoneNumber: Bool, domain: String) async -> (Result<User_BaseResponse, Error>) {
+		var request = User_UpdateProfileRequest()
+		request.avatar = avatar
+		request.displayName = displayName
+		request.phoneNumber = phoneNumber
+		request.clearPhoneNumber_p = clearPhoneNumber
+		return await channelStorage.getChannel(domain: domain).updateProfile(request)
+	}
+	
+	public func changePassword(oldPassword: String, newPassword: String, domain: String) async -> (Result<User_BaseResponse, Error>) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e5e0a713c2a0f396e41d6374bb8f07759fda2936/Packages/ChatSecure/ChatSecure/Classes/Networking/Services/User/UserService.swift#L126"></span>Function body should span 50 lines or less excluding comments and whitespace: currently spans 52 lines<br />
<code>function_body_length</code> <code>UserService.swift:126</code></td>
    </tr>
  </tbody>
</table>